src/deploy/functions/services/auth.ts

```diff
@@ -3,122 +3,148 @@ import * as identityPlatform from "../../../gcp/identityPlatform";
-      !!config.forwardInboundCredentials?.idToken ||
-    !!newConfig.forwardInboundCredentials?.refreshToken !==
-      !!config.forwardInboundCredentials?.refreshToken
+
+  private configChanged(
+    newConfig: identityPlatform.BlockingFunctionsConfig,
+    config: identityPlatform.BlockingFunctionsConfig
   ) {
-    return true;
+    if (
+      newConfig.triggers?.beforeCreate?.functionUri !==
+        config.triggers?.beforeCreate?.functionUri ||
+      newConfig.triggers?.beforeSignIn?.functionUri !== config.triggers?.beforeSignIn?.functionUri
+    ) {
+      return true;
+    }
+    if (
+      !!newConfig.forwardInboundCredentials?.accessToken !==
+        !!config.forwardInboundCredentials?.accessToken ||
+      !!newConfig.forwardInboundCredentials?.idToken !==
+        !!config.forwardInboundCredentials?.idToken ||
+      !!newConfig.forwardInboundCredentials?.refreshToken !==
+        !!config.forwardInboundCredentials?.refreshToken
+    ) {
+      return true;
+    }
+    return false;
   }
-  return false;
-}
 
-/**
- * Registers the auth blocking trigger to identity platform. On updates, we don't touch the options.
- * @param endpoint the blocking endpoint
- * @param update if this registration is an update
- */
-export async function registerBlockingTrigger(
-  endpoint: backend.Endpoint & backend.BlockingTriggered,
-  update: boolean
-): Promise<void> {
-  const newBlockingConfig = await identityPlatform.getBlockingFunctionsConfig(endpoint.project);
-  const oldBlockingConfig = cloneDeep(newBlockingConfig);
+  private async registerTriggerLocked(
+    endpoint: backend.Endpoint & backend.BlockingTriggered
+  ): Promise<void> {
+    const newBlockingConfig = await identityPlatform.getBlockingFunctionsConfig(endpoint.project);
+    const oldBlockingConfig = cloneDeep(newBlockingConfig);
 
-  if (endpoint.blockingTrigger.eventType === events.v1.BEFORE_CREATE_EVENT) {
-    newBlockingConfig.triggers = {
-      ...newBlockingConfig.triggers,
-      beforeCreate: {
-        functionUri: endpoint.uri!,
-      },
-    };
-  } else {
-    newBlockingConfig.triggers = {
-      ...newBlockingConfig.triggers,
-      beforeSignIn: {
-        functionUri: endpoint.uri!,
-      },
-    };
-  }
+    if (endpoint.blockingTrigger.eventType === events.v1.BEFORE_CREATE_EVENT) {
+      newBlockingConfig.triggers = {
+        ...newBlockingConfig.triggers,
+        beforeCreate: {
+          functionUri: endpoint.uri!,
+        },
+      };
+    } else {
+      newBlockingConfig.triggers = {
+        ...newBlockingConfig.triggers,
+        beforeSignIn: {
+          functionUri: endpoint.uri!,
+        },
+      };
+    }
 
-  if (!update) {
     newBlockingConfig.forwardInboundCredentials = {
       idToken: endpoint.blockingTrigger.options?.idToken || false,
       accessToken: endpoint.blockingTrigger.options?.accessToken || false,
       refreshToken: endpoint.blockingTrigger.options?.refreshToken || false,
```

Comment:
I like your intention to preserve OOB changes. I think what I would do is say something like

```
newBlockingConfig.forwardInboundCredentials = { ...oldBlockingConfig.forwardInboundCredentials, ...endpoint.blockingTrigger.options }
```

This would keep options that were set out of band unless it was specified in the customer's code. The `|| false` you have should generally be unnecessary because the API server will automatically fill zero values for any missing field.

Reply:
fixed in #4443